packages/devtools_app/lib/src/shared/preferences.dart

```diff
@@ -64,11 +68,28 @@ class PreferencesController {
 
 class InspectorPreferencesController {
   ValueListenable<bool> get hoverEvalModeEnabled => _hoverEvalMode;
+  ValueListenable<List<String>> get customPubRootDirectories =>
+      _customPubRootDirectories;
+  ValueListenable<bool> get isRefreshingCustomPubRootDirectories =>
+      _customPubRootDirectoriesAreBusy;
+  InspectorService get inspectorService =>
+      serviceManager.inspectorService as InspectorService;
 
   final _hoverEvalMode = ValueNotifier<bool>(false);
+  final _customPubRootDirectories = ListValueNotifier<String>([]);
+  final _customPubRootDirectoriesAreBusy = ValueNotifier<bool>(false);
+  final _busyCounter = ValueNotifier<int>(0);
   static const _hoverEvalModeStorageId = 'inspector.hoverEvalMode';
+  static const _customPubRootDirectoriesStorageId =
+      'inspector.customPubRootDirectories';
 
   Future<void> init() async {
+    await initHoverEvalMode();
+    _initCustomPubRootListeners();
+    setGlobal(InspectorPreferencesController, this);
```

Comment:
why do we need InspectorPreferencesController to be a global when it is accessible from the preferences global? i think we can get rid of this and remove InspectorPreferencesController from globals